packages/nameguard-react/src/components/ReportIcon/index.tsx

```diff
@@ -63,127 +78,98 @@ export function ReportIcon({
   */
   ...props
 }: ReportShieldProps) {
-  const onClickHandler = (clickHandlerFor: ClickHandlerFor) => {
-    switch (clickHandlerFor) {
-      case ClickHandlerFor.icon:
-        if (onIconClickOverride) {
-          onIconClickOverride(ensName);
-        } else {
-          redirectToViewNameReportURL(ensName);
-        }
-        break;
-      case ClickHandlerFor.tooltip:
-        if (onTooltipClickOverride) {
-          onTooltipClickOverride(ensName);
-        } else {
-          redirectToViewNameReportURL(ensName);
-        }
-        break;
+  useEffect(() => {
+    if (data) {
+      if (data.name !== name.name) {
+        throw new Error(
+          `ReportIcon error: The name in the provided data: "${data.name}" does not match the expected name "${name.name}".`,
+        );
+      }
     }
-  };
+  }, [data]);
+
+  let icon: React.ReactNode;
+  let tooltipIcon: React.ReactNode;
+  let tooltipTitleClass: string;
+  let tooltipTitle: string;
+  let tooltipSubtitle: string | undefined = undefined;
+  let tooltipMessage: string | undefined = undefined;
```

Comment:
This is a synonym of:

```suggestion
  let tooltipSubtitle: string | undefined;
  let tooltipMessage: string | undefined;
```

Reply:
Feel free to choose one between two